app/src/main/java/com/andela/omokolataiwo/levelup/view/MainActivity.java

```diff
@@ -32,20 +35,40 @@
      */
     private List<GithubProfile> githubProfilesParcel;
 
-    /**
-     * github profile presenter.
-     */
-    private final GithubProfilePresenter githubProfilePresenter =
-            new GithubProfilePresenter(this);
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
         recyclerView = findViewById(R.id.recycler_view);
+        final SwipeRefreshLayout swipeRefreshLayout = findViewById(
+                R.id.main_activity_swipe_refresh);
+
+        final GithubProfilePresenter githubProfilePresenter = new GithubProfilePresenter(this,
+                (CoordinatorLayout) findViewById(R.id.coordinatorLayout));
+
+        swipeRefreshLayout.setColorSchemeResources(
+                R.color.refresh_red,
+                R.color.refresh_blue,
+                R.color.refresh_green
+        );
+        swipeRefreshLayout.setSize(SwipeRefreshLayout.LARGE);
+        swipeRefreshLayout.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
+            @Override
+            public void onRefresh() {
+                swipeRefreshLayout.setRefreshing(true);
```

Comment:
@omokolataiwo 
Firstly, once the `onRefresh` method is triggered, you do not need to manually call `swipeRefreshLayout.setRefreshing(true)`... It is automatically called.
Secondly, the use of `Handler` here, is quite unnecessary.. I will need to have a one on one sync with you to clarify on that